### ISSUE TITLE
[branch 430] controllers on TP should union the nodes from RP cluster and local cluster

### ISSUE THREAD

**Issue by yb01** [CentaurusInfra/arktos#1042]
**What happened**:
currently when the controller get node, it ONLY get from the RP controllers, in turn, the static pods on local cluster were GCed and cycled ( created-GCed-recreated....)

i.e. this function should also include the local cluster's node:

```
func (gcc *PodGCController) gcOrphaned(pods []*v1.Pod) {
	klog.V(4).Infof("GC'ing orphaned")
	// We want to get list of Nodes from the etcd, to make sure that it's as fresh as possible.

	// get nodes from resource provider clients
	allRpNodes := make(map[string]*v1.NodeList, len(gcc.resourceProviderClients))
	errs := make(map[string]error, len(gcc.resourceProviderClients))
	var wg sync.WaitGroup
	wg.Add(len(gcc.resourceProviderClients))
	var lock sync.Mutex
	for rpId, client := range gcc.resourceProviderClients {
		go func(resourceProviderId string, rpClient clientset.Interface, nodeLists map[string]*v1.NodeList, errs map[string]error, writeLock *sync.Mutex) {
			defer wg.Done()
			nodes, err := rpClient.CoreV1().Nodes().List(metav1.ListOptions{})
			if err != nil {
				writeLock.Lock()
				errs[resourceProviderId] = err
				klog.Errorf("Error listing nodes. err: %v", errs)
				writeLock.Unlock()
				return
			}
			writeLock.Lock()
			nodeLists[resourceProviderId] = nodes
			writeLock.Unlock()
		}(rpId, client, allRpNodes, errs, &lock)
	}
	wg.Wait()

	// check errors and aggregate nodes
	if len(errs) == len(gcc.resourceProviderClients) {
		// avoid garbage collection when
		klog.Errorf("Error listing nodes from all resource partition. err: %v", errs)
		return
	}

	nodeNames := sets.NewString()
	for _, nodes := range allRpNodes {
		for _, node := range nodes.Items {
			nodeNames.Insert(node.Name)
		}
	}

............ 
```
```
I0322 17:04:15.176130       1 gc_controller.go:195] Found orphaned Pod system/kube-system/etcd-server-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.209448       1 gc_controller.go:195] Found orphaned Pod system/kube-system/kube-scheduler-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.218002       1 gc_controller.go:195] Found orphaned Pod system/kube-system/etcd-empty-dir-cleanup-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.224984       1 gc_controller.go:195] Found orphaned Pod system/kube-system/etcd-server-events-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.237561       1 gc_controller.go:195] Found orphaned Pod system/kube-system/etcd-server-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.246049       1 gc_controller.go:195] Found orphaned Pod system/kube-system/kube-controller-manager-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.254343       1 gc_controller.go:195] Found orphaned Pod system/kube-system/kube-addon-manager-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
I0322 17:04:35.263104       1 gc_controller.go:195] Found orphaned Pod system/kube-system/kube-apiserver-yb01-multiple-rp-kubemark-tp-1-master assigned to the Node yb01-multiple-rp-kubemark-tp-1-master. Deleting.
```

**Comment by yb01** [comment]
this is the same as issue 1006.

**Comment by h-w-chen** [comment]
fixed by PR #1044 (in POC 430 branch)
